Library/Homebrew/cmd/commands.rb

```diff
@@ -8,51 +8,76 @@ module Homebrew
   module_function
 
   def commands
-    if ARGV.include? "--quiet"
-      cmds = internal_commands + external_commands
-      cmds += internal_developer_commands
-      cmds += HOMEBREW_INTERNAL_COMMAND_ALIASES.keys if ARGV.include? "--include-aliases"
-      puts Formatter.columns(cmds.sort)
-    else
-      # Find commands in Homebrew/cmd
-      puts "Built-in commands"
-      puts Formatter.columns(internal_commands)
-
-      # Find commands in Homebrew/dev-cmd
-      puts
-      puts "Built-in developer commands"
-      puts Formatter.columns(internal_developer_commands)
-
-      # Find commands in the path
-      unless (exts = external_commands).empty?
-        puts
-        puts "External commands"
-        puts Formatter.columns(exts)
+    # TODO: goal is to no longer require def commands at all
+    CommandsCommand.call
+  end
+
+  class CommandsCommand < Command
+    options do
+      command_name "commands"
+      desc "Show a list of built-in and external commands."
+      option "quiet", switch: "q", desc: "list only the names of commands without the header." do
+        option "include-aliases", switch: "i", desc: "the aliases of internal commands will be included."
       end
+      option "prune", switch: "p", value: "days", desc: "remove all cache files older than <days>."
+      option "prune1", value: "days", desc: "remove all cache files older than <days>."
+      option switch: "f", desc: "do foo"
+      option "l", desc: "lol"
     end
-  end
 
-  def internal_commands
-    find_internal_commands HOMEBREW_LIBRARY_PATH/"cmd"
-  end
+    def self.call
```

Comment:
`call` is what `Proc` and other lambda-ish objects use, so this might get a bit confusing. Maybe `run`, to go along with the language of "running a command"?